test/include/userobjects/ShapeTestUserObject.h

```diff
@@ -0,0 +1,69 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+
+#ifndef SHAPETESTUSEROBJECT_H
+#define SHAPETESTUSEROBJECT_H
+
+#include "ShapeElementUserObject.h"
+
+//Forward Declarations
+class ShapeTestUserObject;
+
+template<>
+InputParameters validParams<ShapeTestUserObject>();
+
+/**
+ * Test and proof of concept class for computing UserObject Jacobians using the
+ * ShapeElementUserObject base class. This object computes the integral
+ * \f$ \int_\Omega u^2v dr \f$
+ * and builds a vector of all derivatives of the integral w.r.t. the DOFs of u and v.
```

Comment:
That being said I could definitely add this. I should probably also write a Jacobian test.